PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/CreateProjectUtil.java

```diff
@@ -183,6 +184,18 @@ public static void addSourceFoldertoClassPath(IJavaProject javaProject, String f
 		
 		javaProject.setRawClasspath(newEntries, new NullProgressMonitor());
 	}
+	
+	public static boolean checkHDInsightProjectNature(NewJavaProjectWizardPageTwo pageTwo) {
+		if (pageTwo != null && pageTwo.getJavaProject() != null && pageTwo.getJavaProject().getProject() != null) {
```

Comment:
You can try `Optonal` class with `map` operation for this in Java8. 😄

Reply:
I will learn those Java8 features:)

---
In reply to: [140948219](https://github.com/Microsoft/azure-tools-for-java/pull/1041#discussion_r140948219) [](ancestors = 140948219)